src/controllers/TorusController.ts

```diff
@@ -1214,6 +1221,11 @@ export default class TorusController extends BaseController<TorusControllerConfi
 
     // sign all transaction
     const allTransactions = req.params?.message?.map((msg) => {
+      if (req.params?.messageOnly) {
+        let tx = Transaction.populate(Message.from(Buffer.from(msg, "hex")));
```

Comment:
messageOnly shouldn't be converted to Transaction for signing. This defeats the whole purpose of using messageOnly. Use nacl.sign in such cases. 
check: https://github.com/solana-labs/solana/issues/23641

Reply:
Fix with signMessage